src/core/ExtensionContext.tsx

```diff
@@ -51,13 +51,12 @@ export function WithExtensionContext({
         };
         const schema = sdk.field.schema;
         const formValue = (await sdk.field.getValue()) as ShoppableImageData;
-        const hasImage = !(formValue as { image: { _empty?: boolean } }).image
-          ?._empty;
+        let hasImage = !!formValue?.image?.id;
         const asset = hasImage
-          ? await sdk.assets.getById(formValue.image.id)
+          ? await sdk.assets.getById(formValue?.image?.id || '')
```

Comment:
could we add hasId to the conditional which should resolve the assets type error aswell 

hasImage && hasId ?